File: src/main/java/com/example/demo/service/NotificationService.java
package com.example.demo.service;

import com.example.demo.dto.request.NotificationCreateRequestDTO;
import com.example.demo.dto.response.NotificationResponseDTO;
import java.util.List;

public interface NotificationService {
    NotificationResponseDTO createNotification(NotificationCreateRequestDTO request);
    NotificationResponseDTO getNotificationById(Long id);
    List<NotificationResponseDTO> getUserNotifications(Long userId);
    List<NotificationResponseDTO> getUnreadNotifications(Long userId);
    void markAsRead(Long id);
    void markAllAsRead(Long userId);
    void deleteNotification(Long id);
}

File: frontend/src/services/notificationService.js
import api from './api';

export const notificationService = {
  getUserNotifications: () => api.get('/notifications/user'),
  getUnreadNotifications: () => api.get('/notifications/user/unread'),
  markAsRead: (id) => api.put(`/notifications/${id}/read`),
  markAllAsRead: () => api.put('/notifications/read-all'),
  deleteNotification: (id) => api.delete(`/notifications/${id}`),
};
